File: Projeto_NFe/Server/Projeto_NFe.Infra.PDF/Features/ShippingCompanies/ShippingCompanyModel.cs
using Projeto_NFe.Domain.Features.ShippingCompanies;
using Projeto_NFe.Domain.Utils;
using Projeto_NFe.Infra.PDF.Features.Addresses;

namespace Projeto_NFe.Infra.PDF.Features.ShippingCompanies
{
    public class ShippingCompanyModel
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public string CorporateName { get; set; }
        public string Cnpj { get; set; }
        public string Cpf { get; set; }
        public string StateRegistration { get; set; }
        public bool ShippingResponsability { get; set; }
        public AddressModel Address { get; set; }
        public EnumPersonType PersonType { get; set; }

        public void Create(ShippingCompany shippingCompany)
        {
            Cnpj = shippingCompany.Cnpj;
            Cpf = shippingCompany.Cpf;
            Address = new AddressModel();
            Address.Neighborhood = shippingCompany.Address.Neighborhood;
            Address.State = shippingCompany.Address.State;
            Address.StreetName = shippingCompany.Address.StreetName;
            Address.City = shippingCompany.Address.City;
            Address.Number = shippingCompany.Address.Number;
            Address.Country = shippingCompany.Address.Country;
            Id = shippingCompany.Id;
            StateRegistration = shippingCompany.StateRegistration;
            Name = shippingCompany.BusinessName;
            CorporateName = shippingCompany.CorporateName;
            ShippingResponsability = shippingCompany.ShippingResponsability;
            PersonType = shippingCompany.PersonType;
        }
    }
}

File: Client/src/app/features/shipping-company/shared/shipping-company.model.ts
import { Address } from '../../address/address.model';

export class ShippingCompany {
    public id: number;
    public businessName: string;
    public corporateName: string;
    public cnpj: string;
    public cpf: string;
    public personType: number;
    public stateRegistration: string;
    public streetName: string;
    public number: number;
    public neighborhood: string;
    public city: string;
    public state: string;
    public country: string;
}

export class ShippingCompanyRegisterCommand {
    public businessName: string;
    public corporateName: string;
    public cnpj: string;
    public cpf: string;
    public personType: number;
    public stateRegistration: string;
    public streetName: string;
    public number: number;
    public neighborhood: string;
    public city: string;
    public state: string;
    public country: string;

    constructor(shippingCompany: ShippingCompany, address: Address) {
        this.businessName = shippingCompany.businessName;
        this.corporateName = shippingCompany.corporateName;
        this.cnpj = shippingCompany.cnpj;
        this.cpf = shippingCompany.cpf;
        this.stateRegistration = shippingCompany.stateRegistration;
        this.personType = shippingCompany.personType;
        this.streetName = address.streetName;
        this.number = address.number;
        this.neighborhood = address.neighborhood;
        this.city = address.city;
        this.state = address.state;
        this.country = address.country;
    }
}

export class ShippingCompanyUpdateCommand {
    public id: number;
    public businessName: string;
    public corporateName: string;
    public cnpj: string;
    public cpf: string;
    public personType: number;
    public stateRegistration: string;
    public streetName: string;
    public number: number;
    public neighborhood: string;
    public city: string;
    public state: string;
    public country: string;

    constructor(shippingCompany: ShippingCompany, address: Address, id: number) {
        this.id = id;
        this.businessName = shippingCompany.businessName;
        this.corporateName = shippingCompany.corporateName;
        this.cnpj = shippingCompany.cnpj;
        this.cpf = shippingCompany.cpf;
        this.stateRegistration = shippingCompany.stateRegistration;
        this.personType = shippingCompany.personType;
        this.streetName = address.streetName;
        this.number = address.number;
        this.neighborhood = address.neighborhood;
        this.city = address.city;
        this.state = address.state;
        this.country = address.country;
    }
}

export class ShippingCompanyRemoveCommand {
    public ids: number[] = [];

    constructor(selectedEntities: any) {
        this.ids = selectedEntities.map((shippingCompany: ShippingCompany) => shippingCompany.id);
    }
}
